tests/suite/known_triples.rs

```diff
@@ -0,0 +1,92 @@
+use std::collections::BTreeSet;
+
+use platforms::Platform;
+
+#[test]
+fn gen_known_triples() {
+    let output = "src/dist/triple/known.rs";
+    let existing = std::fs::read_to_string(output).unwrap();
+    assert_eq!(known_triples(), existing);
+}
+
+/// Generates the lists of known target architectures, OSes and environments.
+fn known_triples() -> String {
```

Comment:
I would prefer to just inline the contents of this function, since it only has a single caller anyway.